### PR TITLE
Tweaks to evse test plans based on review(issue #31460)

### LINKED ISSUE #31460: Tweaks to EVSE Test plans based on review #30957
### Reproduction steps

- [x] Move the class EventChangeCallback into matter_test_support.py
- [x] Update send_enable_charge_command not to have min_charge/max_charge default to None
- [ ] Update the charging duration test to have 120s or a better value (may need a test plan change)
- [x] Add CI checking run

### Bug prevalence

N/A

### GitHub hash of the SDK that was being used

N/A see PR #31200

### Platform

core

### Platform Version(s)

1.3

### Type

Test Improvement

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jamesharrow**
Fixes #31460 

[x] - Add EVSE tests to .github/workflow/tests.yaml
[x] - Move Test event trigger support and Event classes to matter_testing_support for more widespread use
[x] - Update send_enable_charge_command not to have min_charge/max_charge default to None

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -503,6 +504,9 @@ jobs:
...
+                  scripts/run_in_python_env.sh out/venv './scripts/tests/run_python_test.py --app out/linux-x64-energy-management-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-energy-management-app  --factoryreset --app-args "--discriminator 1234 --KVS kvs1 --trace-to json:out/trace_data/app-{SCRIPT_BASE_NAME}.json --enable-key 000102030405060708090a0b0c0d0e0f" --script "src/python_testing/TC_EEVSE_2_2.py" --script-args "--storage-path admin_storage.json --commissioning-method on-network --discriminator 3840 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --trace-to json:out/trace_data/test-{SCRIPT_BASE_NAME}.json --trace-to perfetto:out/trace_data/test-{SCRIPT_BASE_NAME}.perfetto"'
```
[the remark this answers is not in the document]
Comment (jamesharrow) [reply .github/workflows/tests.yaml]: Fixed - hadn't seen that there two `--discriminators` in the veryyyyy long command line. :( 
Thanks

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -503,6 +504,9 @@ jobs:
...
+                  scripts/run_in_python_env.sh out/venv './scripts/tests/run_python_test.py --app out/linux-x64-energy-management-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-energy-management-app  --factoryreset --app-args "--discriminator 1234 --KVS kvs1 --trace-to json:out/trace_data/app-{SCRIPT_BASE_NAME}.json --enable-key 000102030405060708090a0b0c0d0e0f" --script "src/python_testing/TC_EEVSE_2_4.py" --script-args "--storage-path admin_storage.json --commissioning-method on-network --discriminator 3840 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --trace-to json:out/trace_data/test-{SCRIPT_BASE_NAME}.json --trace-to perfetto:out/trace_data/test-{SCRIPT_BASE_NAME}.perfetto"'
```
[the remark this answers is not in the document]
Comment (jamesharrow) [reply .github/workflows/tests.yaml]: Fixed

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -503,6 +504,9 @@ jobs:
...
+                  scripts/run_in_python_env.sh out/venv './scripts/tests/run_python_test.py --app out/linux-x64-energy-management-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-energy-management-app  --factoryreset --app-args "--discriminator 1234 --KVS kvs1 --trace-to json:out/trace_data/app-{SCRIPT_BASE_NAME}.json --enable-key 000102030405060708090a0b0c0d0e0f" --script "src/python_testing/TC_EEVSE_2_5.py" --script-args "--storage-path admin_storage.json --commissioning-method on-network --discriminator 3840 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --trace-to json:out/trace_data/test-{SCRIPT_BASE_NAME}.json --trace-to perfetto:out/trace_data/test-{SCRIPT_BASE_NAME}.perfetto"'
```
[the remark this answers is not in the document]
Comment (jamesharrow) [reply .github/workflows/tests.yaml]: Fixed

File: examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp [hunk 4]
```diff
@@ -232,8 +232,8 @@ void EVSEManufacturer::FakeReadingsUpdate()
 
     // Update meter values
     // Avoid using floats - so we will do a basic rand() call which will generate a integer value between 0 and RAND_MAX
-    // first compute power as a mean + some random value in range 0 to mPowerRandomness_mW
-    int64_t power = (rand() % gFakeReadingsData.mPowerRandomness_mW);
+    // first compute power as a mean + some random value in range +/- mPowerRandomness_mW
+    int64_t power = ((int64_t) rand() % (2 * gFakeReadingsData.mPowerRandomness_mW)) - gFakeReadingsData.mPowerRandomness_mW;
```
Comment (cecille) [inline examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp:236]: please use C++ style cast - static_cast<int64_t>(rand()). Not sure if mPowerRandomness is already a 64

**Review by cecille — APPROVED** [review]
Thanks, James - this is very helpful to have in the base.

File: src/python_testing/matter_testing_support.py [hunk 5]
```diff
@@ -847,6 +876,35 @@ async def send_single_cmd(
         result = await dev_ctrl.SendCommand(nodeid=node_id, endpoint=endpoint, payload=cmd, timedRequestTimeoutMs=timedRequestTimeoutMs)
         return result
 
+    async def send_test_event_triggers(self, enableKey: bytes = None, eventTrigger: int = 0x0099000000000000):
```
Comment (tcarmelveilleux) [inline src/python_testing/matter_testing_support.py:879]: Please do not default to a particular cluater

File: src/python_testing/matter_testing_support.py [hunk 6]
```diff
@@ -847,6 +876,35 @@ async def send_single_cmd(
...
+                                       cmd=Clusters.GeneralDiagnostics.Commands.TestEventTrigger(
+                                           enableKey,
+                                           eventTrigger)
+                                       )
+
+        except InteractionModelError as e:
+            asserts.fail(
+                f"Sending TestEventTrigger resulted in Unexpected error. Are they enabled in DUT? Command returned - {e.status}")
+
+    async def check_test_event_triggers_enabled(self):
```
Comment (tcarmelveilleux) [inline src/python_testing/matter_testing_support.py:901]: Please add docstrings to all new methods in this common module

File: src/python_testing/matter_testing_support.py [hunk 7]
```diff
@@ -273,6 +274,34 @@ def name(self) -> str:
...
+                                                      fabricFiltered=False, keepSubscriptions=True, autoResubscribe=False)
+        self._subscription.SetEventUpdateCallback(self.__call__)
+
+    def __call__(self, res: EventReadResult, transaction: SubscriptionTransaction):
+        if res.Status == Status.Success and res.Header.ClusterId == self._expected_cluster.id:
+            logging.info(
+                f'Got subscription report for event on cluster {self._expected_cluster}: {res.Data}')
+            self._q.put(res)
+
+    def WaitForEventReport(self, expected_event: ClusterObjects.ClusterEvent, timeout: int = 10):
```
Comment (tcarmelveilleux) [inline src/python_testing/matter_testing_support.py:294]: Please use pep8 snake_case style for methods, arguments and local variables

**Review by tcarmelveilleux — APPROVED** [review]

File: src/python_testing/matter_testing_support.py (hunk 6 above)
Comment (jamesharrow) [reply src/python_testing/matter_testing_support.py:901]: Added

File: src/python_testing/matter_testing_support.py (hunk 5 above)
Comment (jamesharrow) [reply src/python_testing/matter_testing_support.py:879]: Reordered params and removed.

File: src/python_testing/matter_testing_support.py (hunk 7 above)
Comment (jamesharrow) [reply src/python_testing/matter_testing_support.py:294]: Changed

File: examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp (hunk 4 above)
Comment (jamesharrow) [reply examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp:236]: Changed

**Review by arkq — APPROVED** [review]

**Pull request merged**